### PR TITLE
Frontend scaffolding (header and footer)

### PR DESCRIPTION
**Pull request by vivianliang**
screen shot 2015-11-24 at 12 18 53 pm

### CODE REVIEW

File: appfiles/app/views/directives/topbar.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<div class="topbar navbar navbar-default" role="navigation">
+  <div class="container">
+    <div class="navbar-header">
+      <a class="navbar-brand" href="#/">downe.space</a>
+    </div>
+
+    <a
+      class="btn btn-sm btn-default"
+      ng-class="{'disabled': user === null}"
```
Comment (vivianliang) [inline appfiles/app/views/directives/topbar.html]: wip

File: appfiles/app/styles/views/topbar.less [hunk 2]
```diff
@@ -0,0 +1,35 @@
+.navbar-default .navbar-brand {
+  color: #83abb9;            // light-blue
```
Comment (vivianliang) [inline appfiles/app/styles/views/topbar.less]: @variables not working

Comment (mgeist) [inline appfiles/app/styles/views/topbar.less]: there might be something weird going on with the less compilation. i was noticing it was breaking whenever i ran grunt test also :(

File: appfiles/app/views/directives/topbar.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<div class="topbar navbar navbar-default" role="navigation">
```
Comment (vivianliang) [inline appfiles/app/views/directives/topbar.html]: * might be fine to just remove these bootstrap classes
* include a `min-width` on this

File: appfiles/app/styles/views/topbar.less [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  &:active, &:focus {
+    color: #83abb9;          // light-blue
+  }
+}
+
+.topbar {
+  background-color: #184060; // dark-blue
+  border-color    : #184060; // dark-blue
+  border-radius   : 0;
+  // min-width       : 1000px;
```
Comment (vivianliang) [inline appfiles/app/styles/views/topbar.less]: remove

File: appfiles/app/styles/views/footer.less [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  height          : 60px;
+  bottom          : 0;
+  background-color: #184060; // dark-blue
+  border-top      : 1px solid #e5e5e5;
+  text-align      : center;
+
+  a {
+    color: #83abb9;            // light-blue
+    padding-right: 10px;
+    &:hover {
```
Comment (vivianliang) [inline appfiles/app/styles/views/footer.less]: nit: space

File: appfiles/app/styles/main.less [hunk 6]
```diff
@@ -7,38 +9,22 @@
...
 body {
   padding: 0;
+  margin-bottom: 60px; // footer height
 }
 
 /* Everything but the jumbotron gets side spacing for mobile first views */
-.header,
-.marketing,
-.footer {
+.marketing {
```
Comment (vivianliang) [inline appfiles/app/styles/main.less]: get rid of these classes. and jumbotron below

File: appfiles/app/views/directives/topbar.html [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+    <form class="navbar-form navbar-left" role="search">
+      <div class="form-group">
+        <input type="text" class="form-control" placeholder="City or Zip Code">
+      </div>
+      <button type="submit" class="btn btn-default">Search</button>
+    </form>
+    <a
+      class="btn btn-md btn-default btn-submit-event"
+      ng-class="{'disabled': user === null}"
```
Comment (mgeist) [inline appfiles/app/views/directives/topbar.html]: check the codebase for currentUser, should be able to hook this up legit

File: appfiles/app/scripts/directives/footer.coffee [hunk 8]
```diff
@@ -0,0 +1,12 @@
+'use strict'
+
+angular.module('downespace').directive 'footer', ->
+  directive =
+    controller  : 'topBarController'
```
Comment (mgeist) [inline appfiles/app/scripts/directives/footer.coffee]: nit these could use renaming to footerController etc (line below and controller too)

Comment (vivianliang) [inline appfiles/app/scripts/directives/footer.coffee]: lulz

File: appfiles/app/styles/main.less (hunk 6 above)
Comment (vivianliang) [inline appfiles/app/styles/main.less]: nvm leaving for now in case example is helpful for figuring out mobile styling

**Pull request merged**